hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/has/isauthorizedraw/IsAuthorizedRawCall.java

```diff
@@ -90,27 +95,38 @@ public IsAuthorizedRawCall(
     public PricedResult execute(@NonNull final MessageFrame frame) {
         requireNonNull(frame, "frame");
 
-        final Function<ResponseCodeEnum, PricedResult> bail = rce -> reversionWith(rce, HARDCODED_GAS_REQUIREMENT_GAS);
-
-        // Fail fast if user didn't supply monstrous amount of gas
-        final long availableGas = frame.getRemainingGas();
-        if (availableGas < HARDCODED_GAS_REQUIREMENT_GAS) return bail.apply(INSUFFICIENT_GAS);
-
-        // Figure out what kind of signature we're dealing with thus what kind of account+key we need
+        // First things first: What kind of signature are we dealing with, thus what kind of account+key we need
         final var signatureType =
                 switch (signature.length) {
                     case 65 -> SignatureType.EC;
                     case 64 -> SignatureType.ED;
                     default -> SignatureType.INVALID;
                 };
+
+        // Now we know how much gas this call will cost
+        final long gasRequirement =
+                switch (signatureType) {
+                    case EC -> HARDCODED_EC_GAS_REQUIREMENT_GAS;
+                    case ED -> HARDCODED_ED_GAS_REQUIREMENT_GAS;
+                    case INVALID -> Math.min(HARDCODED_EC_GAS_REQUIREMENT_GAS, HARDCODED_ED_GAS_REQUIREMENT_GAS);
+                };
+
+        // Prepare the short-circuit error status returns
+        final Function<ResponseCodeEnum, PricedResult> bail = rce -> reversionWith(rce, gasRequirement);
+
+        // Must have a valid signature type to continue
         if (signatureType == SignatureType.INVALID)
             return bail.apply(INVALID_TRANSACTION_BODY /* should be: "invalid argument to precompile" */);
 
+        // Fail immediately if user didn't supply sufficient gas
+        final long availableGas = frame.getRemainingGas();
```

Comment:
I think we can remove this, as its already performed in CustomMessageCallProcessor.doExecuteSystemContract, right after IsAuthorizedRawCall.execute() has been called.